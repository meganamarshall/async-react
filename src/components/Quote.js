import React from 'react';
import PropTypes from 'prop-types';

function Quote({ quote }) {
  return (
    <section>
      <p>{quote.character}</p>
      <p>{quote.quote}</p>
      <img src={quote.image} />
    </section> 
  );
}

Quote.propTypes = {
  quote: PropTypes.shape({
    character: PropTypes.string.isRequired,
    quote: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  }).isRequired
};

export default Quote;
