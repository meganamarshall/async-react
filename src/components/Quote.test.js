import React from 'react';
import { shallow } from 'enzyme';
import Quote from './Quote';

describe('quote component', () => {
  it('renders the quote', () => {
    const quote = {
      character: 'Megan',
      quote: 'Hi',
      image: 'imagepath.png'
    };
    const wrapper = shallow(<Quote quote={quote} />);
    expect(wrapper).toMatchSnapshot();
  });
});
