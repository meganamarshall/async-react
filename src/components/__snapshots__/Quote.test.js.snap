// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quote component renders the quote 1`] = `
<section>
  <p>
    Megan
  </p>
  <p>
    Hi
  </p>
  <img
    src="imagepath.png"
  />
</section>
`;
